test/Microsoft.Azure.Relay.UnitTests/WebSocketTests.cs

```diff
@@ -201,6 +180,51 @@ async Task<bool> TestAcceptHandler(RelayedHttpListenerContext listenerContext)
             return result;
         }
 
+        async Task TestRawWebSocket(HybridConnectionListener listener, EndpointTestType endpointTestType)
+        {
+            var clientWebSocket = new ClientWebSocket();
+            var wssUri = await GetWebSocketConnectionUri(endpointTestType);
+
+            using (var cancelSource = new CancellationTokenSource(TimeSpan.FromMinutes(1)))
+            {
+                TestUtility.Log($"Opening {listener}");
+                await listener.OpenAsync(cancelSource.Token);
+
+                await clientWebSocket.ConnectAsync(wssUri, cancelSource.Token);
+
+                var listenerStream = await listener.AcceptConnectionAsync();
+
+                TestUtility.Log("Client and Listener are connected!");
+                Assert.Null(clientWebSocket.SubProtocol);
+
+                byte[] sendBuffer = this.CreateBuffer(1024, new byte[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 });
+                await clientWebSocket.SendAsync(new ArraySegment<byte>(sendBuffer), WebSocketMessageType.Binary, true, cancelSource.Token);
+                TestUtility.Log($"clientWebSocket wrote {sendBuffer.Length} bytes");
+
+                byte[] readBuffer = new byte[sendBuffer.Length];
+                await this.ReadCountBytesAsync(listenerStream, readBuffer, 0, readBuffer.Length, TimeSpan.FromSeconds(30));
+                Assert.Equal(sendBuffer, readBuffer);
+
+                TestUtility.Log("Calling clientStream.CloseAsync");
+                var clientStreamCloseTask = clientWebSocket.CloseAsync(WebSocketCloseStatus.EndpointUnavailable, "From Test Code", cancelSource.Token);
+                TestUtility.Log("Reading from listenerStream");
+                int bytesRead = await this.SafeReadAsync(listenerStream, readBuffer, 0, readBuffer.Length);
+                TestUtility.Log($"listenerStream.Read returned {bytesRead} bytes");
+                Assert.Equal(0, bytesRead);
+
+                TestUtility.Log("Calling listenerStream.CloseAsync");
+                var listenerStreamCloseTask = listenerStream.CloseAsync(cancelSource.Token);
+                await listenerStreamCloseTask;
+                TestUtility.Log("Calling listenerStream.CloseAsync completed");
+                await clientStreamCloseTask;
+                TestUtility.Log("Calling clientStream.CloseAsync completed");
+
+                TestUtility.Log($"Closing {listener}");
+                await listener.CloseAsync(TimeSpan.FromSeconds(10));
+                listener = null;
```

Comment:
>listener = null; [](start = 16, length = 16)

Since there's no longer a finally block which might also close this listener setting to null no longer helps anything.

Reply:
The listener is passed to this method. You can see the finally block in the test methods itself.

Reply:
That's the thing, this was getting set to null so that the finally block didn't bother trying to cleanup the listener. Setting it to null inside this method is now useless.

---
In reply to: [605401417](https://github.com/Azure/azure-relay-dotnet/pull/145#discussion_r605401417) [](ancestors = 605401417)

Reply:
okay makes sense will remove the listener = null .

Reply:
done